tests/console/validate_ext4_fs.pm

```diff
@@ -0,0 +1,91 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate ext4 filesystem and system partitioning.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+
+    select_console "root-console";
+
+    my $test_data  = get_test_suite_data();
+    my @partitions = @{$test_data->{partitions}};
+    my $size       = 0;
+    my $tmp;
+
+    foreach (@partitions) {
+
+        if ($_->{mnt_point} eq '[SWAP]') {
+
+            # Validate that the partition is created.
+
+            record_info("Check $_->{mnt_point}", "Verify the '$_->{mnt_point}' partition was created.");
+            assert_script_run("lsblk -n | grep SWAP", fail_message => "Partition SWAP was not created.");
+
+        }
+        else {
+
+            # Validate that the partition is created.
+
+            record_info("Check $_->{mnt_point}", "Verify the '$_->{mnt_point}' partition was created.");
+            assert_script_run("lsblk -n | grep $_->{mnt_point}\$", fail_message => "Partition $_->{mnt_point} was not created.");
+
+            # Validate that the filesystem type is the expected one.
+
+            record_info("Check fs $_->{mnt_point}", "Verify the '$_->{mnt_point}' partition filesystem is '$_->{fs_type}'");
+            my $fstype = script_output("df -PT $_->{mnt_point} | grep -v \"Filesystem\" | awk '{print \$2}'");
+            assert_equals($_->{fs_type}, $fstype, "File system on '$_->{mnt_point}' partition does not correspond to the expected one.");
+
+            # Test the ability to read and write files on the partition.
+
+            assert_script_run("echo Hello > $_->{mnt_point}/emptyfile", fail_message => 'Failure while writing in ' . $_->{mnt_point});
+            assert_script_run("cat $_->{mnt_point}/emptyfile",          fail_message => 'Failure while reading from ' . $_->{mnt_point});
+
+        }
+
+    }
+
+    # Validate that there is no unpartitioned space.
+
+
+    for (my $i = 1; $i <= ((scalar @{$test_data->{partitions}}) + 1); $i++) {
+
+        $tmp = script_output("fdisk -l | grep \"/dev/vda$i\" | awk '{print \$5}'");
```

Comment:
shouldn't be some direct command to check unpartitioned space (or not yet allocated to a partition). Did you try `parted /dev/vda unit GiB print free` or `df -h`. Those addition looks error-prone.